### PR TITLE
Get rid of some linting errors/warnings

### PR DESCRIPTION
**Pull request by cgccuser**
A couple resources weren't closed, and a couple interfaces were redundantly implemented. That's about it.

### CODE REVIEW

**Review by lyxal — APPROVED** [review]
Lgtm

File: src/main/java/io/github/seggan/jyxal/runtime/ProgramStack.java [hunk 1]
```diff
@@ -4,10 +4,9 @@
 
 import java.util.ArrayDeque;
 import java.util.Collection;
-import java.util.Deque;
 import java.util.List;
 
-public class ProgramStack extends ArrayDeque<Object> implements Deque<Object> {
+public class ProgramStack extends ArrayDeque<Object> {
```
Comment (Seggan) [inline src/main/java/io/github/seggan/jyxal/runtime/ProgramStack.java]: I like being explicit with these

File: src/main/java/io/github/seggan/jyxal/runtime/RuntimeHelpers.java [hunk 2]
```diff
@@ -47,10 +47,12 @@ public static JyxalList deepCopy(JyxalList list) {
 
     public static Object exec(String expr) {
         ProgramStack stack = new ProgramStack();
-        for (SnippetEvent e : jShell.get().eval(jShell.get().sourceCodeAnalysis().analyzeCompletion(expr).source())) {
+        var shell = jShell.get();
```
Comment (Seggan) [inline src/main/java/io/github/seggan/jyxal/runtime/RuntimeHelpers.java]: Please don't use `var`

File: src/test/java/io/github/seggan/jyxal/TestHelper.java [hunk 3]
```diff
@@ -26,6 +23,7 @@ public static void run(String... args) throws IOException {
             Class<?> clazz = cl.loadClass("jyxal.Main");
             clazz.getMethod("main", String[].class).invoke(null, (Object) new String[0]);
         } catch (ReflectiveOperationException e) {
+            cl.close();
```
Comment (Seggan) [inline src/test/java/io/github/seggan/jyxal/TestHelper.java]: If you want it this way, put it in a `finally` block and remove both `close()` calls

**Review by Seggan — CHANGES_REQUESTED** [review]
I'm feeling quite picky today 😉

File: src/main/java/io/github/seggan/jyxal/runtime/math/internal/BigDecimalMath.java [hunk 4]
```diff
@@ -27,16 +27,9 @@ of this software and associated documentation files (the "Software"), to deal
 import java.math.BigDecimal;
 import java.math.MathContext;
 import java.math.RoundingMode;
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
+import java.util.*;
```
Comment (Seggan) [inline src/main/java/io/github/seggan/jyxal/runtime/math/internal/BigDecimalMath.java:30]: No star imports please

File: src/main/java/io/github/seggan/jyxal/runtime/math/internal/BigDecimalMath.java [hunk 5]
```diff
@@ -27,16 +27,9 @@ of this software and associated documentation files (the "Software"), to deal
...
 
-import static java.math.BigDecimal.ONE;
-import static java.math.BigDecimal.TEN;
-import static java.math.BigDecimal.ZERO;
-import static java.math.BigDecimal.valueOf;
+import static java.math.BigDecimal.*;
```
Comment (Seggan) [inline src/main/java/io/github/seggan/jyxal/runtime/math/internal/BigDecimalMath.java:32]: Same here

File: src/main/java/io/github/seggan/jyxal/runtime/math/internal/BigDecimalMath.java [hunk 6]
```diff
@@ -1077,42 +1070,40 @@ else if (value < 10.0) { // (8.5 - 9.0 - 10.0) -> (0.94444 - 1.0 - 1.11111)
             powerOfThree = 9;
         }
         else {
-            while (value > 1.4) { // never happens when called by logUsingExponent()
-                value /= 2;
-                factorOfTwo++;
-	            powerOfTwo <<= 1;
-            }
-        }
+					while (value > 1.4) { // never happens when called by logUsingExponent()
```
Comment (Seggan) [inline src/main/java/io/github/seggan/jyxal/runtime/math/internal/BigDecimalMath.java:1073]: Spaces, not tabs

**Review by Seggan — CHANGES_REQUESTED** [review]
I should've run refactor on the entire project lol

**Pull request closed without merging**